addons/website_sale/models/sale_order.py

```diff
@@ -360,6 +356,24 @@ def action_recovery_email_send(self):
             },
         }
 
+    @api.multi
+    def _get_recovery_template(self):
+        try:
+            default_template = self.env.ref('website_sale.mail_template_sale_cart_recovery', raise_if_not_found=False)
+            template = (self.filtered('website_id') == self and
+                        self.mapped('website_id')[-1:1].cart_recovery_mail_template_id.id) or default_template
+        except:
+            template = False
+        return template
+
+    @api.multi
+    def recovery_email_send(self):
+        for order in self:
+            order._portal_ensure_token()
+            template = order._get_recovery_template()
+            if template:
+                template.with_context(website_sale_send_recovery_email=True).send_mail(order.id, force_send=False, raise_exception=False)
```

Comment:
I believe those are the default values for `force_send` and `raise_exception` so no need to specify them.

If we intend to have the override of `send_mail` in `website_sale` to work, setting `website_sale_send_recovery_email` without also having `active_ids` will not trigger the condition.
But since this is in a `TransientModel` for `mail.compose.message` and that we don't want to compose the message here, we probably don't need to change the context at all?